lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2796,8 +2796,8 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
       return false;
     for (auto paramPair : llvm::zip(candidateFnTy.getParams(),
                                     required.getParams()))
-      if (std::get<0>(paramPair).getParameterType() !=
-          std::get<1>(paramPair).getParameterType())
+      if (!std::get<0>(paramPair).getPlainType()->isEqual(
```

Comment:
Explanation: `getParameterType()` canonicalizes variadic parameter types to become array types: `Int32... -> [Int32]`. `getPlainType()` doesn't.